vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -199,15 +199,13 @@ window.Vaadin.Flow.gridConnector = {
         return;
       }
       if (!newVal) {
-        if (oldVal && selectedKeys[oldVal.key]) {
+    	if (!grid.$connector.deselectAllowed && oldVal) {
+    	  grid.activeItem = oldVal;
+    	} else if (oldVal && selectedKeys[oldVal.key]) {
           grid.$connector.doDeselection([oldVal], true);
         }
-        return;
-      }
-      if (!selectedKeys[newVal.key]) {
+      } else if (!selectedKeys[newVal.key]) {
```

Comment:
There are some tabs in the indentation. Change to spaces.

Reply:
Fixed